src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -325,8 +326,11 @@ private void OnQueryButtonClick(object sender, RoutedEventArgs e)
 		}
 
 		private void SubmitSearch()
-		{
-			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(_suggestionsList.SelectedItem, Text));
+	    {	
+			string finalResult = GetObjectText(Text);
+
+			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(finalResult is "" ? null : finalResult, userInput));			
```

Comment:
I'm sorry but I'm going to be annoying and ask for more clarification :) It is very possible that I am just not understanding but wouldn't this not solve the underlying issue?

According to the [AutoSuggestBox official docs](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.autosuggestbox?view=windows-app-sdk-1.1#query-submitted) the `ChosenSuggestion` object should only be set if an item was actually selected from the list, otherwise, if I type "aaa" and hit Enter, the QuerySubmitted EventArgs should have a null `ChosenSuggestion` and a `QueryText` of "aaa". 

> The QuerySubmitted event occurs when a user commits a query string. The user can commit a query in one of these ways:
> - While focus is in the text box, press Enter or click the query icon. The event args [ChosenSuggestion](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.autosuggestboxquerysubmittedeventargs.chosensuggestion?view=windows-app-sdk-1.1#microsoft-ui-xaml-controls-autosuggestboxquerysubmittedeventargs-chosensuggestion) property is null.
> - While focus is in the suggestion list, press Enter, click, or tap an item. The event args [ChosenSuggestion](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.autosuggestboxquerysubmittedeventargs.chosensuggestion?view=windows-app-sdk-1.1#microsoft-ui-xaml-controls-autosuggestboxquerysubmittedeventargs-chosensuggestion) property contains the item that was selected from the list. In all cases, the event args [QueryText](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.autosuggestboxquerysubmittedeventargs.querytext?view=windows-app-sdk-1.1#microsoft-ui-xaml-controls-autosuggestboxquerysubmittedeventargs-querytext) property contains the text from the text box.

Looking at this code, it seems we are always setting the `ChosenSuggestion` to be the query string.

Also, I don't think we can always assume that `ChosenSuggestion` is a string. The `AutoSuggestBox` is an `ItemsControl` so the `ChosenSuggestion` should be the same type as the `ItemsSource`.